docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+| pmon on NPU     | Verify when pmon crash on NPU.          | • Start  I/O<br>• Kill pmon on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+| bgpd on NPU     | Verify when bgpd crash on NPU.          | • Start  I/O<br>• Kill bgpd on NPU        | DPU1 remains active, DPU2 remains standby.        | T2 receives packets without disruption.          |
+
+###  Module 7 Power down and hardware failure
+For each case in this module, there are 2 variations:
+1. Failure happens on DPU1, traffic landing on DPU2
+2. Failure happens on DPU2, traffic landing on DPU1
+
+| Case                   | Goal                                             | Test Steps                                           | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|------------------------|--------------------------------------------------|------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| DPU hardware failure   | Verify traffic flow when DPU hardware fails     | • Start  I/O<br>• Force DPU reset (ChassisStateDB DPU_STATE) | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets without disruption.          |
+| T1 unplanned reboot    | Verify traffic when T1 ungracefully reboots     | • Start  I/O<br>• Reboot T1               | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets without disruption.          |
+| T1 power down          | Verify traffic when T1 power down               | • Start  I/O<br>• Toggle T1 PDU link     | DPU1 becomes non-active, DPU2 becomes standalone. | T2 receives packets without disruption.          |
+
+
+### Module 8 Operations
+For each case in this module, there are 2 variations:
+1. Failure happens on DPU1, traffic landing on DPU2
+2. Failure happens on DPU2, traffic landing on DPU1
+
+| Case                                   | Goal                                                    | Test Steps                                                                   | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|----------------------------------------|---------------------------------------------------------|------------------------------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| Shutdown/Startup BGP sessions from NPU | Verify traffic when shutdown and startup sessions from NOS | • Start  I/O<br>• Shutdown all BGP sessions on NPU<br>• Startup all BGP sessions on NPU | Impacted side become non-active, the peer side become standalone. | T2 receives packets without disruption.          |
```

Comment:
I believe these will cause packet drops too, maybe I missed something or maybe the scenario that we are intended to test here is not described clearly.

Reply:
Addressed.